Source/BoundaryConditions/WarpX_PEC.cpp

```diff
@@ -333,33 +333,44 @@ namespace
 
 
     /**
-     * \brief Sets the rho or J field value in cells close to and on reflecting particle boundary
-     *        or PEC field boundary. The charge/current density deposited
-     *        in the guard cells are either reflected
-     *        back into the simulation domain (if a reflecting particle
-     *        boundary is used), or the opposite charge/current density is deposited
-     *        back in the domain to capture the effect of an image charge.
-     *        The charge/current density on the reflecting boundary is set to 0 while values
-     *        in the guard cells are set equal (and opposite) to their mirror
-     *        location inside the domain - representing image charges - in the
-     *        normal (tangential) direction.
+     * \brief Sets the rho or J field value near PEC and PMC boundaries by appropriately
+     *        folding in the rho or J deposited into the ghost cells.
+     *        Note that energy conservation demands that the boundary conditions for J be
+     *        consistent with those for E used to push the particles. If E_i is odd, then
+     *        J_i deposited to ghost cells should be subtracted from its mirror location
+     *        inside the domain. If E_i is even, then J_i deposited to the ghost cells should
+     *        be added to its mirror location inside the domain. If this is not done,
```

Comment:
I think these lines can be removed since the PEC and PMC subheadings below explains clearly what happens in each case. Plus the "odd" or "even" distinction for E_i is not clear to me. Is "odd" = parallel to boundary?
```suggestion
     *        consistent with those for E used to push the particles. If this is not done,
```

Reply:
Oh I see it is explained in the doc-string for `ApplyBoundarytoJfield` below.